### PR TITLE
add java to pmmlserver, testing instructions

### LINKED ISSUE #30: pmmlserver rock missing required dependency java
### Bug Description

On trying to integrate the pmmlserver into the kserve controler for https://github.com/canonical/kserve-operators/issues/208 it was found that the model server requires java be installed.  We missed this I think because:
* the pmmlserver only tries to use java once it has a model.  We have no integration tests in this rock repo, so likely we did not actually mount a model and try executing it during manual tests
* its easy to miss in the [upstream dockerfile](https://github.com/kserve/kserve/blob/release-0.11/python/pmml.Dockerfile) - they build [`FROM` an image that includes java](https://github.com/kserve/kserve/blob/b688db6be4e640cfbc89030dd802428652da58c0/python/pmml.Dockerfile#L3C5-L3C15)

### To Reproduce

We can manually reproduce this by:

```
# Install the gcloud CLI tool: https://cloud.google.com/sdk/docs/install

# download the model locally
gsutil cp -r gs://kfserving-examples/models/pmml ./sample_model

# Build the rock and put it to docker
tox -e pack
tox -e export-to-docker

# mount the model into the container at runtime
docker run -v PATH_TO/sample_model:/mnt/models PMMLSERVER_IMAGE --model_name pmml-demo --model_dir=/mnt/models/pmml --http_port=8080
```

### Environment

-

### Relevant Log Output

```shell
-
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ca-scribner**
closes #30 

## Testing instructions

See `pmmlserver/README.md`

### CODE REVIEW

File: pmmlserver/rockcraft.yaml [hunk 1]
```diff
@@ -29,6 +29,16 @@ parts:
...
 
+  java:
+    plugin: nil
+    build-packages:
+      - default-jdk
+    stage-packages:
+      - default-jdk
+    # Also stage the links for java executable
+    override-build:
+      ln -s /usr/lib/jvm/java-11-openjdk-amd64/bin/java $CRAFT_PART_INSTALL/usr/bin/java
```
Comment (DnPlas) [inline pmmlserver/rockcraft.yaml:40]: Were you hitting something like `Error: java-11-openjdk-amd64/bin/java not found or not a valid JDK`?

**Review by DnPlas — COMMENTED** [review]
Thanks @ca-scribner ! Just a tiny question, other than that LGTM.

File: pmmlserver/rockcraft.yaml (hunk 1 above)
Comment (ca-scribner) [reply pmmlserver/rockcraft.yaml:40]: yeah the error was something like you mentioned.  doing:

```
    build-packages:
      - default-jdk
    stage-packages:
      - default-jdk
```

stages the binary, but it does not bring any of the symlinks for `java` that are in the users path

**Review by DnPlas — APPROVED** [review]

File: pmmlserver/rockcraft.yaml (hunk 1 above)
Comment (DnPlas) [reply pmmlserver/rockcraft.yaml:40]: Alrighty, LGTM!

File: pmmlserver/README.md
[the remark this answers is not in the document]
Comment (ca-scribner) [reply pmmlserver/README.md:1]: we sorted this out in #38 to remove the duplication

**Pull request merged**